src/lib/spinel/spinel.h

```diff
@@ -321,11 +321,47 @@
  *
  *   - SPINEL_MIN_HOST_SUPPORTED_RCP_API_VERSION specifies the minimum spinel
  *     RCP API Version which is supported by the host-side implementation.
+ *     To reduce the backward compatibility issues, this number should be kept
+ *     as constant as possible.
  *
  *   - On start, host implementation queries the RCP API version and accepts
  *     any version number from SPINEL_MIN_HOST_SUPPORTED_RCP_API_VERSION up to
  *     and including SPINEL_RCP_API_VERSION.
  *
+ *   Host and RCP compatibility guideline:
+ *
+ *   - New host spinel layer should work with an older RCP firmware, i.e., host
+ *     implementation should remain backward compatible.
+ *
+ *   - Existing fields in the format of an already implemented spinel
+ *     property or command cannot change.
+ *
+ *   - New fields should be appended to the end of the existing spinel format.
+ *     *  New fields for new features:
+ *          Adding a new capability flag to the otRadioCaps to indicate the new
```

Comment:
Does it mean we need to add a new capability flag whenever we are extending a spinel format?

The host should already be able to recognize the new format by checking its length?

I think `otRadioCaps` are for larger scope functionalities that may or may not be supported by the device, sometimes due to hardware restrictions. 

Thoughts?

Reply:
When we are extending the new fields for a new feature, we need to add a new capability flag. Otherwise, we should void adding new fields.

We didn't stipulate that when a new feature is enabled, the old feature must be enabled too. If some old features are disabled, the host can't recognize the new format by checking its length.

I agree that `otRadioCaps` should be for larger scope functionalities. We should try our best to avoid adding new fields for non-feature changes.